File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {  
    public boolean isSubtree(TreeNode root, TreeNode subRoot) {
        return dfs(root,subRoot);
    }
    public boolean dfs(TreeNode root, TreeNode subRoot){
        if(root!=null && subRoot!=null && root.val==subRoot.val && isSameTree(root,subRoot)){
            return true;
        }
        if(root!=null){
            return dfs(root.left,subRoot) || dfs(root.right,subRoot);
        }
        return false;
    }

        public boolean isSameTree(TreeNode p, TreeNode q) {
        if(q==null && p==null){
            return true;
        }
        if( p==null || q==null ||p.val!=q.val ){
            return false;
        }

        return isSameTree(p.left,q.left) && isSameTree(p.right,q.right);

    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        curr_max=[0]
        def dfs(root,i):
            if not root:
                return 
            dfs(root.left,i+1)
            dfs(root.right,i+1)
            curr_max[0]=max(curr_max[0],i+1)
        dfs(root,0)
        return curr_max[0]